app/main.py

```diff
@@ -2,0 +1,51 @@
+class SoftwareEngineer:
+    skills = []
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    skills = ["JavaScript", "HTML", "CSS"]
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+class BackendDeveloper(SoftwareEngineer):
+    skills = ["Python", "SQL", "Django"]
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+class AndroidDeveloper(SoftwareEngineer):
+    skills = ["Java", "Android studio"]
```

Comment:
The `skills` attribute in `AndroidDeveloper` is a class variable. This will be shared across all instances of `AndroidDeveloper`. Consider moving it to the constructor to make it an instance variable.